File: backend/app/schemas/resume.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ResumeUploadResponse(BaseModel):
    id: int
    filename: str
    original_filename: str
    file_type: str
    file_size: int
    created_at: datetime


class ResumeHistoryItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    original_filename: str
    file_type: str
    file_size: int
    created_at: datetime


class ResumeHistoryResponse(BaseModel):
    items: list[ResumeHistoryItem]

File: frontend/src/types/resume.ts
export interface ResumeUploadResponse {
  id: number;
  filename: string;
  original_filename: string;
  file_type: string;
  file_size: number;
  created_at: string;
}

export interface ResumeHistoryItem {
  id: number;
  original_filename: string;
  file_type: string;
  file_size: number;
  created_at: string;
}

export interface ResumeHistoryResponse {
  items: ResumeHistoryItem[];
}
